lib/coach/handler.rb

```diff
@@ -25,16 +25,28 @@ def call(env)
       start = Time.now
 
       publish('coach.handler.start', start_event.dup)
-      response = chain.instrument.call
 
-      finish = Time.now
-      publish('coach.handler.finish',
-              start, finish, nil,
-              start_event.
-                merge(response: { status: response[0] },
-                      metadata: context.fetch(:_metadata, {})))
+      begin
+        response = chain.instrument.call
+      rescue => e
+        finish = Time.now
+        publish('coach.handler.error',
+                start, finish, nil,
+                start_event.merge(
+                  response: { status: 500 },
```

Comment:
Do we have to introduce a new Coach event?

If we think of this as the Coach application layer, then as far as Coach is concerned a 500 is the appropriate response. If a (Rack) middleware further up the stack happens to modify that status and/or manipulate the response in some way, then we'd be reporting incorrect statuses for successful Coach requests too.

Following that logic, I think it's fair to say at the point the application errors, Coach has generated a 500.

```ruby
def call(env)
  ...
  begin do
    response = chain.instrument.call # don't catch exceptions from this
  ensure
    finish = Time.now
    publish(
      'coach.handler.request',
      start, finish, nil,
      start_event.merge(
        response: { status: response&.first || 500 },
        metadata: context.fetch(:_metadata, {})
      )
    )
  end
end
```

I think something like above makes more sense. You make sure that whatever the chain does, you publish the event, with either the response from the chain or a 500. It doesn't make sense to me that people would need an extra notification event (`coach.handler.error`) to be instrumenting when things go wrong. It should be sufficient to filter for 500 errors on requests.

Does this make sense?

---

On an aside, we do need the response so that we can generate the `coach.request` meta-event. Removing the response would break the presentation of that event, which would be sad.